govcd/openapi.go

```diff
@@ -251,19 +271,27 @@ func (client *Client) OpenApiPostItemAsync(apiVersion string, urlRef *url.URL, p
 // asynchronous requests. The urlRef must point to POST endpoint (e.g. '/1.0.0/edgeGateways'). When a task is
 // synchronous - it will track task until it is finished and pick reference to marshal outType.
 func (client *Client) OpenApiPostItem(apiVersion string, urlRef *url.URL, params url.Values, payload, outType interface{}, additionalHeader map[string]string) error {
+	_, err := client.OpenApiPostItemAndGetHeader(apiVersion, urlRef, params, payload, outType, additionalHeader, "")
+	return err
+}
+
+// OpenApiPostItemAndGetHeader is a low level OpenAPI client function to perform POST request for item supporting synchronous or
+// asynchronous requests, that returns also the given header value. The urlRef must point to POST endpoint (e.g. '/1.0.0/edgeGateways'). When a task is
+// synchronous - it will track task until it is finished and pick reference to marshal outType.
+func (client *Client) OpenApiPostItemAndGetHeader(apiVersion string, urlRef *url.URL, params url.Values, payload, outType interface{}, additionalHeader map[string]string, headerToRetrieve string) (string, error) {
```

Comment:
Similar comment to the other functions - could it return all headers so that it is more flexible for other user cases

Reply:
Same, refactored. Thanks for the suggestion, it is a good improvement indeed.